### PR TITLE
retire blocks: pre-check if db has enough data to build files

### PR DESCRIPTION
**Pull request by AskAlexSharov**
this PR is preventing error `err="DumpBlocks: DumpBodies: header missed`

### CODE REVIEW

File: turbo/snapshotsync/freezeblocks/block_snapshots.go [hunk 1]
```diff
@@ -1174,13 +1177,50 @@ func CanDeleteTo(curBlockNum uint64, blocksInSnapshots uint64) (blockTo uint64)
...
+		lastInFiles := br.snapshots().SegmentsMax() + 1
+		haveGap = lastInFiles < firstInDB
+		if haveGap {
+			log.Debug("[snapshots] not enuogh blocks in db to create snapshots", "lastInFiles", lastInFiles, " firstBlockInDB", firstInDB, "recommendations", "it's ok to ignore this message. can fix by: downloading more files `rm datadir/snapshots/prohibit_new_downloads.lock datdir/snapshots/snapshots-lock.json`, or downloading old blocks to db `integration stage_headers --reset`")
+		}
+		return nil
+	}); err != nil {
+		return false, err
+	}
+	if haveGap {
```
Comment (yperbasis) [inline turbo/snapshotsync/freezeblocks/block_snapshots.go:1200]: can be simplified to `return !haveGap, nil`

File: turbo/snapshotsync/freezeblocks/block_snapshots.go [hunk 2]
```diff
@@ -1174,13 +1177,50 @@ func CanDeleteTo(curBlockNum uint64, blocksInSnapshots uint64) (blockTo uint64)
...
+		if err != nil {
+			return err
+		}
+		if !ok {
+			return nil
+		}
+		lastInFiles := br.snapshots().SegmentsMax() + 1
+		haveGap = lastInFiles < firstInDB
+		if haveGap {
+			log.Debug("[snapshots] not enuogh blocks in db to create snapshots", "lastInFiles", lastInFiles, " firstBlockInDB", firstInDB, "recommendations", "it's ok to ignore this message. can fix by: downloading more files `rm datadir/snapshots/prohibit_new_downloads.lock datdir/snapshots/snapshots-lock.json`, or downloading old blocks to db `integration stage_headers --reset`")
```
Comment (yperbasis) [inline turbo/snapshotsync/freezeblocks/block_snapshots.go:1194]: typo in "enuogh"

File: turbo/snapshotsync/freezeblocks/bor_snapshots.go [hunk 3]
```diff
@@ -30,15 +28,57 @@ import (
 	"github.com/ledgerwatch/erigon/eth/ethconfig"
 	"github.com/ledgerwatch/erigon/polygon/heimdall"
 	"github.com/ledgerwatch/erigon/turbo/services"
+	"github.com/ledgerwatch/log/v3"
 )
 
+func (br *BlockRetire) dbHasEnoughDataForBorRetire(ctx context.Context) (bool, error) {
+	/*
```
Comment (yperbasis) [inline turbo/snapshotsync/freezeblocks/bor_snapshots.go:35]: Why is this commented out?

Comment (AskAlexSharov) [reply turbo/snapshotsync/freezeblocks/bor_snapshots.go:35]: bor-events exist not in all blocks. don't know which check to add here yet.

**Review by awskii — APPROVED** [review]

**Pull request merged**